packages/@aws-cdk/aws-eks/lib/cluster-resource-handler/common.ts

```diff
@@ -14,6 +14,22 @@ export interface EksUpdateId {
 
 export type ResourceEvent = AWSLambda.CloudFormationCustomResourceEvent & EksUpdateId;
 
+/**
+ * Decodes encoded true/false values
+ */
+function decodeBooleans(object: object) {
```

Comment:
Where does this come into play? Who uses these values to represent booleans?

Reply:
it was the only way i found to pass down booleans explicitly to the handler, so that i could test modifying a previously created EKS cluster with something like this:

```ts
    const clusterResource = cluster.node.defaultChild as cdk.CfnResource;
    const clusterCustomResource = clusterResource.node.defaultChild as cfn.CfnCustomResource;
    const cfnCluster = clusterCustomResource.node.defaultChild as cdk.CfnResource;
    cfnCluster.addPropertyOverride('Config.logging.clusterLogging', [
      {
        types: [ 'api' ],
        enabled: 'TRUE:BOOLEAN',
      }
    ]);
```

as a workaround (and possibly help the implementation later) for https://github.com/aws/aws-cdk/issues/4159

Reply:
What happened if you just passed in `true` or `false`?

Reply:
the custom resource by default converts every input to string, so the call to `updateClusterConfig` is sent with 'true' or 'false' as string and then obviously it fails because it expects a boolean

Reply:
Yes, these booleans are passed as strings but they propagate all the way to the SDK and AFAIU it "just works". Why doesn't it fail when users pass `true`/`false` it in production or do we have a latent issue here?

Reply:
i believe it is a latent issue and people are not facing more problems because booleans are not being used for EKS specifically (or because the other API calls are handling booleans as strings just fine)

Reply:
either way if we can make things more explicit the better, i would propose to have an additional option encoding the custom resource properties into these special strings.
https://github.com/aws/aws-cdk/blob/01ef1ca9818b2bd9f219de04ce2ec657de4e2149/packages/%40aws-cdk/core/lib/custom-resource.ts#L112-L113

if you agree i could create a separate PR just for that

Reply:
@eduardomourar sounds interesting (and makes more sense to solve at a lower layer), but let's get this PR merged.

I still don't understand what happens if you remove the `decodeBooleans` function. What impact does it have on the user?

Reply:
for now none.

to avoid confusion i will move `decodeBooleans` to a new PR in order to implement #4159
and to deal with booleans in the custom resources while at it. does it make sense?

Reply:
Sounds good. That's where my confusion came from.

Reply:
Thank you!